src/MainComponent.hpp

```diff
@@ -17,7 +17,8 @@ using namespace ftxui;
 class MainComponent : public ComponentBase
 {
  public:
-  MainComponent(Receiver<std::string> receiverRunOutput, Receiver<ErrorMessage> receiverErrorOutput);
+  MainComponent(Receiver<std::string> receiverRunOutput, Receiver<ErrorMessage> receiverErrorOutput, Component runButton, Component quitButton,
+                std::atomic<int>* progress);
```

Comment:
Move more of the functionality inside MainComponent by moving runButton quitButton and run E+ gauge inside of it